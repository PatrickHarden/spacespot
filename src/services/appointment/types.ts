export interface AppointmentState {
  data?: Array<Appointment>
  error?: string
}

export interface Appointment {
  id: number
  enquireId: number
  description: string
  peer1Id: string
  peer2Id: string
  proposedTime: string
  createdAt: string
  lastUpdatedAt: string
  lastProposedBy: string
  appointmentStatus: string
}

export interface NewAppointment {
  description: string
  proposedTime: string
}
export interface EditAppointment {
  id: string
  description: string
  proposedTime: string
}
export interface AppointmentAction {
  type: string
  payload?: Array<Appointment> | NewAppointment | EditAppointment | string
}
