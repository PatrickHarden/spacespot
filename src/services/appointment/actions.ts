import {
  AppointmentAction,
  Appointment,
  EditAppointment,
  NewAppointment,
} from './types'
import constants from './constants'

const get = (): AppointmentAction => ({
  type: constants.GET,
})
const getSuccess = (payload: Array<Appointment>) => ({
  type: constants.GET_SUCCESS,
  payload,
})
const getError = (payload: string): AppointmentAction => ({
  type: constants.GET_ERROR,
  payload,
})
const update = (): AppointmentAction => ({
  type: constants.UPDATE,
})

const create = (payload: NewAppointment): AppointmentAction => ({
  type: constants.CREATE,
  payload,
})

const createSuccess = (payload: Appointment) => ({
  type: constants.CREATE_SUCCESS,
  payload,
})
const createError = (payload: string): AppointmentAction => ({
  type: constants.CREATE_ERROR,
  payload,
})

const accept = (payload: string): AppointmentAction => ({
  type: constants.ACCEPT,
  payload,
})

const acceptSuccess = (payload: Appointment) => ({
  type: constants.ACCEPT_SUCCESS,
  payload,
})

const acceptError = (payload: string): AppointmentAction => ({
  type: constants.ACCEPT_ERROR,
  payload,
})

const cancel = (payload: string): AppointmentAction => ({
  type: constants.CANCEL,
  payload,
})

const cancelSuccess = (payload: Appointment) => ({
  type: constants.CANCEL_SUCCESS,
  payload,
})

const cancelError = (payload: string): AppointmentAction => ({
  type: constants.CANCEL_ERROR,
  payload,
})

const change = (payload: EditAppointment): AppointmentAction => ({
  type: constants.CHANGE,
  payload,
})

const changeSuccess = (payload: Appointment) => ({
  type: constants.CHANGE_SUCCESS,
  payload,
})

const changeError = (payload: string): AppointmentAction => ({
  type: constants.CHANGE_ERROR,
  payload,
})
export default {
  get,
  getSuccess,
  getError,
  update,
  create,
  createSuccess,
  createError,
  accept,
  acceptSuccess,
  acceptError,
  cancel,
  cancelSuccess,
  cancelError,
  change,
  changeSuccess,
  changeError,
}
