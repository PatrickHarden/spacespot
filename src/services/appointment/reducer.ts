import constants from './constants'
import { AppointmentAction, AppointmentState, Appointment } from './types'

const initialState: AppointmentState = {}

const reducer = (
  state: AppointmentState = initialState,
  action: AppointmentAction,
): AppointmentState => {
  switch (action.type) {
    case constants.GET_ERROR:
      return { ...state, error: action.payload as string }
    case constants.GET_SUCCESS:
      return { ...state, data: action.payload as Array<Appointment> }
    case constants.CREATE_SUCCESS:
      return {
        ...state,
        data: [...(state.data || []), action.payload as Appointment],
      }
    default:
      return state
  }
}
export default reducer
