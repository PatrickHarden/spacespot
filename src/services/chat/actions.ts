import { ChatAction, ChatResponse, ChatMessage } from './types'
import constants from './constants'

const get = (): ChatAction => ({
  type: constants.GET,
})
const getSuccess = (payload: ChatResponse) => ({
  type: constants.GET_SUCCESS,
  payload,
})
const getError = (payload: string): ChatAction => ({
  type: constants.GET_ERROR,
  payload,
})

const newMessage = (payload: string) => ({
  type: constants.NEW_MESSAGE,
  payload,
})
const newMessageSuccess = (payload: ChatMessage) => ({
  type: constants.NEW_MESSAGE_SUCCESS,
  payload,
})
const newMessageError = (payload: string) => ({
  type: constants.NEW_MESSAGE_ERROR,
  payload,
})
export default {
  get,
  getSuccess,
  getError,
  newMessage,
  newMessageSuccess,
  newMessageError,
}
