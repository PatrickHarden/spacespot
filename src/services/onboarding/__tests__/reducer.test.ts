import reducer from '../reducer'
import actions from '../actions'
import {
  OnBoardingState,
  AvailabilityFlex,
  SpaceStatus,
  AmenityOptions,
} from '../types'
import { Space } from '../../space/types'
import DashboardBuildingSpace from '../../space/__mocks__/dashboard-building-space.json'

let state: OnBoardingState
const mockDate = new Date('2020-01-01')
const amenities: AmenityOptions = {
  hasLifter: { desc: 'Lifert', checked: false },
  hasSecurity: { desc: 'Security', checked: false },
}
let unk: unknown

const availability: AvailabilityFlex = {
  desks: 20,
  availableFrom: mockDate,
  minLease: 2,
  price: 2000,
}

describe('Onboarding reducer', () => {
  it('should return the initial state', () => {
    state = reducer(state, actions.setName('name1'))
    state.spaces[0].availabilityFixed = mockDate
    state.spaces[0].fixedDesks.availableFrom = mockDate
    state.spaces[0].hotDesks.availableFrom = mockDate
    state.spaces[0].servicedOffices.availableFrom = mockDate
    expect(state).toMatchSnapshot()
    state = reducer(state, actions.setDescription('desc1'))
    expect(state).toMatchSnapshot()
    state = reducer(state, actions.setAddress('addres1'))
    state = reducer(state, actions.setAmenities(unk as AmenityOptions))
    state = reducer(state, actions.setAmenities(amenities))
    state = reducer(state, actions.setAspects(unk as Array<string>))
    state = reducer(state, actions.setAspects(['hasLifter']))
    state = reducer(state, actions.setAspects(['none']))
    state = reducer(state, actions.setBuildingId('id1'))
    state = reducer(state, actions.setCity('Olso'))
    state = reducer(state, actions.setPostCode('Olso'))
    state = reducer(state, actions.setLocation('Olso'))
    state = reducer(state, actions.setDescription('descr1'))
    state = reducer(state, actions.setIsNew(true))
    state = reducer(state, actions.setLatLng({ lat: 0, lng: 0 }))
    state = reducer(state, actions.setImages([]))
    state = reducer(state, actions.setDocs([]))
    state = reducer(state, actions.setSpaceDescription('0', 'space1'))
    state = reducer(state, actions.setSpaceFloor('0', 1))
    state = reducer(state, actions.setSpaceFloored('0', 'data'))
    state = reducer(state, actions.setSpaceHighlights('0', 'data'))
    state = reducer(state, actions.setSpaceStatus('0', SpaceStatus.New))
    state = reducer(state, actions.setSpaceMatterPort('0', 'data'))
    state = reducer(state, actions.setSpaceMonths('0', 1))
    state = reducer(state, actions.setSpaceName('0', 'space1'))
    state = reducer(state, actions.setSpaceRent('0', 5000))
    state = reducer(state, actions.setSpaceServices('0', 500))
    state = reducer(state, actions.setSpaceSize('0', 500))
    state = reducer(state, actions.setSpaceSizeCommon('0', 550))
    state = reducer(state, actions.setSpaceType('0', 'FLEX'))
    state = reducer(state, actions.cloneSpace('0'))
    state = reducer(state, actions.setSpaceFitout('0', []))
    state = reducer(state, actions.setSpaceImages('0', []))
    state = reducer(state, actions.setSpaceFloorPlan('0', []))
    state = reducer(state, actions.setSpaceAvailabilityFixed('0', mockDate))
    state = reducer(state, actions.setSpaceFixedDesks('0', availability))
    state = reducer(state, actions.setSpaceServicedOffices('0', availability))
    state = reducer(state, actions.setSpaceHotDesks('0', availability))
    state = reducer(state, actions.setSpaceUse('0', 'OFFICE'))
    state = reducer(state, actions.setSpaceServicesNotNegotiable('0', true))
    state = reducer(state, actions.setIsPublishing(true))
    state = reducer(
      state,
      actions.setDefaultAmenities({
        hasGym: {
          checked: true,
          desc: 'hasGym',
        },
      }),
    )
    state = reducer(state, actions.setCountry('NO'))
    state = reducer(state, actions.setUploadedDocs([]))
    state = reducer(state, actions.setUploadedImages([]))
    state = reducer(state, actions.setSpaceUploadedImages('0', []))
    state = reducer(state, actions.setSpaceUploadedFloorPlans('0', []))
    state = reducer(state, actions.setError('Error1'))
    expect(state).toMatchSnapshot()
    state = reducer(state, actions.resetSpaces())
    state = reducer(state, actions.addNewSpace())
    expect(state).toMatchSnapshot()
    state = reducer(state, actions.removeSpace('0'))
    expect(state).toMatchSnapshot()
  })

  describe('Onboarding reducer with one edit building', () => {
    it('should return the edited building', () => {
      state = reducer(state, actions.clear())
      expect(state).toMatchSnapshot()
      const building: Space = (DashboardBuildingSpace.dashboard
        .buildings[0] as unknown) as Space
      state = reducer(state, actions.onboardingEditBuilding(building))
      expect(state).toMatchSnapshot()
    })
  })
  describe('Onboarding reducer with one edit space', () => {
    it('should return the edited space', () => {
      state = reducer(state, actions.clear())

      const space: Space = (DashboardBuildingSpace.dashboard.spaces[
        'SS_BUILDING_325'
      ][0] as unknown) as Space

      state = reducer(state, actions.onboardingEditSpace(space))
      expect(state).toMatchSnapshot()
    })
  })
})
