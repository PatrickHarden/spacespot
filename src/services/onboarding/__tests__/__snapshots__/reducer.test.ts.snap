// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Onboarding reducer Onboarding reducer with one edit building should return the edited building 1`] = `
Object {
  "address": "",
  "amenities": Object {
    "hasLifter": Object {
      "checked": false,
      "desc": "Lifert",
    },
    "hasSecurity": Object {
      "checked": false,
      "desc": "Security",
    },
    "none": Object {
      "checked": false,
      "desc": "none",
    },
  },
  "buildingId": undefined,
  "city": "",
  "country": "",
  "defaultAmenities": Object {
    "hasGym": Object {
      "checked": true,
      "desc": "hasGym",
    },
  },
  "description": "",
  "district": "",
  "docs": Array [],
  "images": Array [],
  "isNew": undefined,
  "isPublishing": false,
  "latLng": undefined,
  "location": "",
  "name": "",
  "postCode": "",
  "spaceCount": 1,
  "spaces": Object {
    "0": Object {
      "availabilityFixed": 2020-01-01T00:00:00.000Z,
      "fitout": Array [],
      "fixedDesks": Object {
        "availableFrom": 2020-01-01T00:00:00.000Z,
        "currencyCode": "NOK",
        "desks": 0,
        "frequency": "Monthly",
        "minLease": 0,
        "price": 0,
      },
      "floorPlan": Array [],
      "hotDesks": Object {
        "availableFrom": 2020-01-01T00:00:00.000Z,
        "currencyCode": "NOK",
        "desks": 0,
        "frequency": "Monthly",
        "minLease": 0,
        "price": 0,
      },
      "id": "0",
      "images": Array [],
      "months": undefined,
      "servicedOffices": Array [],
      "spaceDescription": "",
      "spaceFloor": undefined,
      "spaceFloored": "",
      "spaceHighlights": "",
      "spaceMatterPort": "",
      "spaceName": "",
      "spaceRent": undefined,
      "spaceServices": undefined,
      "spaceServicesNotNegotiable": false,
      "spaceSize": undefined,
      "spaceSizeCommon": undefined,
      "status": 0,
      "type": "",
      "use": "OFFICE",
    },
  },
  "subdistrict": Object {
    "id": 0,
    "name": "",
  },
}
`;

exports[`Onboarding reducer Onboarding reducer with one edit building should return the edited building 2`] = `
Object {
  "address": "Bryggegata, Oslo, Noruega",
  "amenities": Object {
    "hasGym": Object {
      "checked": true,
      "desc": "hasGym",
    },
    "hasReception-Manned": Object {
      "checked": true,
      "desc": "hasReception-Manned",
    },
  },
  "buildingId": "SS_BUILDING_325",
  "city": "Oslo",
  "country": "Norway",
  "defaultAmenities": Object {
    "hasGym": Object {
      "checked": true,
      "desc": "hasGym",
    },
  },
  "description": "building description",
  "district": "",
  "docs": Array [],
  "images": Array [],
  "isNew": false,
  "isPublishing": false,
  "latLng": Object {
    "lat": 59.91041,
    "lng": 0,
  },
  "location": "",
  "name": "building headline",
  "postCode": "28004",
  "spaceCount": 0,
  "spaces": Object {},
  "subdistrict": Object {
    "id": 0,
    "name": "",
  },
  "uploadedDocs": Array [],
  "uploadedImages": Array [
    Object {
      "Common.ImageCaption": "11 Strand 1_Photo_1_small.jpg",
      "Common.ImageResources": Array [
        Object {
          "Common.Breakpoint": "original",
          "Common.Image.Height": 426,
          "Common.Image.Width": 640,
          "Common.Resource.Uri": "/resources/fileassets/SS_BUILDING_325/5cd79f56/11%20Strand%201_Photo_1_small.jpg",
          "Source.Uri": "https://uatlistingssearchcbreeuw.blob.core.windows.net/fileassets/SS_BUILDING_325/5cd79f56/11%20Strand%201_Photo_1_small.jpg",
        },
        Object {
          "Common.Breakpoint": "small",
          "Common.Image.Height": 426,
          "Common.Image.Width": 640,
          "Common.Resource.Uri": "/resources/fileassets/SS_BUILDING_325/5cd79f56/11%20Strand%201_Photo_1_small_Photo_1_small.jpg",
          "Source.Uri": "https://uatlistingssearchcbreeuw.blob.core.windows.net/fileassets/SS_BUILDING_325/5cd79f56/11%20Strand%201_Photo_1_small.jpg",
        },
        Object {
          "Common.Breakpoint": "medium",
          "Common.Image.Height": 852,
          "Common.Image.Width": 1280,
          "Common.Resource.Uri": "/resources/fileassets/SS_BUILDING_325/5cd79f56/11%20Strand%201_Photo_1_small_Photo_1_medium.jpg",
          "Source.Uri": "https://uatlistingssearchcbreeuw.blob.core.windows.net/fileassets/SS_BUILDING_325/5cd79f56/11%20Strand%201_Photo_1_small.jpg",
        },
        Object {
          "Common.Breakpoint": "large",
          "Common.Image.Height": 1065,
          "Common.Image.Width": 1600,
          "Common.Resource.Uri": "/resources/fileassets/SS_BUILDING_325/5cd79f56/11%20Strand%201_Photo_1_small_Photo_1_large.jpg",
          "Source.Uri": "https://uatlistingssearchcbreeuw.blob.core.windows.net/fileassets/SS_BUILDING_325/5cd79f56/11%20Strand%201_Photo_1_small.jpg",
        },
      ],
    },
  ],
}
`;

exports[`Onboarding reducer Onboarding reducer with one edit space should return the edited space 1`] = `
Object {
  "address": "Bryggegata, Oslo, Noruega",
  "amenities": Object {
    "hasGym": Object {
      "checked": true,
      "desc": "hasGym",
    },
  },
  "buildingId": "SS_BUILDING_325",
  "city": "Oslo",
  "country": "Norway",
  "defaultAmenities": Object {
    "hasGym": Object {
      "checked": true,
      "desc": "hasGym",
    },
  },
  "description": "",
  "district": "",
  "docs": Array [],
  "images": Array [],
  "isNew": false,
  "isPublishing": false,
  "latLng": Object {
    "lat": 59.91041,
    "lng": 0,
  },
  "location": "",
  "name": "Space headline",
  "postCode": "28004",
  "spaceCount": 0,
  "spaces": Object {
    "SS_SPACE_326": Object {
      "address": "Bryggegata, Oslo, Noruega",
      "availabilityFixed": 2019-12-18T00:00:00.000Z,
      "fitout": Array [
        Object {
          "amount": 12,
          "description": "head desc fit1",
          "id": "fitout",
          "isCustom": false,
          "name": "head fitout 1",
        },
      ],
      "fixedDesks": undefined,
      "floorPlan": Array [],
      "hotDesks": undefined,
      "id": "SS_SPACE_326",
      "images": Array [],
      "months": undefined,
      "servicedOffices": Array [],
      "spaceDescription": "",
      "spaceFloor": 0,
      "spaceFloored": "",
      "spaceHighlights": "Space headline",
      "spaceMatterPort": "http://www.matterports.es",
      "spaceName": "Space headline",
      "spaceRent": 21,
      "spaceServices": 1200,
      "spaceServicesNotNegotiable": true,
      "spaceSize": 500,
      "spaceSizeCommon": 500,
      "status": 2,
      "type": "FIXED",
      "uploadedFloorPlans": Array [],
      "uploadedImages": Array [
        Object {
          "Common.ImageCaption": "40 Villiers Street.External_Photo_1_small.jpg",
          "Common.ImageResources": Array [
            Object {
              "Common.Breakpoint": "original",
              "Common.Image.Height": 492,
              "Common.Image.Width": 640,
              "Common.Resource.Uri": "/resources/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small.jpg",
              "Source.Uri": "https://uatlistingssearchcbreeuw.blob.core.windows.net/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small.jpg",
            },
            Object {
              "Common.Breakpoint": "small",
              "Common.Image.Height": 492,
              "Common.Image.Width": 640,
              "Common.Resource.Uri": "/resources/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small_Photo_1_small.jpg",
              "Source.Uri": "https://uatlistingssearchcbreeuw.blob.core.windows.net/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small.jpg",
            },
            Object {
              "Common.Breakpoint": "medium",
              "Common.Image.Height": 984,
              "Common.Image.Width": 1280,
              "Common.Resource.Uri": "/resources/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small_Photo_1_medium.jpg",
              "Source.Uri": "https://uatlistingssearchcbreeuw.blob.core.windows.net/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small.jpg",
            },
            Object {
              "Common.Breakpoint": "large",
              "Common.Image.Height": 1230,
              "Common.Image.Width": 1600,
              "Common.Resource.Uri": "/resources/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small_Photo_1_large.jpg",
              "Source.Uri": "https://uatlistingssearchcbreeuw.blob.core.windows.net/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small.jpg",
            },
          ],
        },
      ],
      "use": "OFFICE",
    },
  },
  "subdistrict": Object {
    "id": 0,
    "name": "",
  },
  "uploadedDocs": Array [],
  "uploadedImages": Array [
    Object {
      "Common.ImageCaption": "40 Villiers Street.External_Photo_1_small.jpg",
      "Common.ImageResources": Array [
        Object {
          "Common.Breakpoint": "original",
          "Common.Image.Height": 492,
          "Common.Image.Width": 640,
          "Common.Resource.Uri": "/resources/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small.jpg",
          "Source.Uri": "https://uatlistingssearchcbreeuw.blob.core.windows.net/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small.jpg",
        },
        Object {
          "Common.Breakpoint": "small",
          "Common.Image.Height": 492,
          "Common.Image.Width": 640,
          "Common.Resource.Uri": "/resources/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small_Photo_1_small.jpg",
          "Source.Uri": "https://uatlistingssearchcbreeuw.blob.core.windows.net/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small.jpg",
        },
        Object {
          "Common.Breakpoint": "medium",
          "Common.Image.Height": 984,
          "Common.Image.Width": 1280,
          "Common.Resource.Uri": "/resources/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small_Photo_1_medium.jpg",
          "Source.Uri": "https://uatlistingssearchcbreeuw.blob.core.windows.net/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small.jpg",
        },
        Object {
          "Common.Breakpoint": "large",
          "Common.Image.Height": 1230,
          "Common.Image.Width": 1600,
          "Common.Resource.Uri": "/resources/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small_Photo_1_large.jpg",
          "Source.Uri": "https://uatlistingssearchcbreeuw.blob.core.windows.net/fileassets/SS_SPACE_326/97117227/40%20Villiers%20Street.External_Photo_1_small.jpg",
        },
      ],
    },
  ],
}
`;

exports[`Onboarding reducer should return the initial state 1`] = `
Object {
  "address": "",
  "amenities": Object {},
  "buildingId": undefined,
  "city": "",
  "country": "",
  "description": "",
  "district": "",
  "docs": Array [],
  "images": Array [],
  "isNew": undefined,
  "isPublishing": false,
  "latLng": undefined,
  "location": "",
  "name": "name1",
  "postCode": "",
  "spaceCount": 1,
  "spaces": Object {
    "0": Object {
      "availabilityFixed": 2020-01-01T00:00:00.000Z,
      "fitout": Array [],
      "fixedDesks": Object {
        "availableFrom": 2020-01-01T00:00:00.000Z,
        "currencyCode": "NOK",
        "desks": 0,
        "frequency": "Monthly",
        "minLease": 0,
        "price": 0,
      },
      "floorPlan": Array [],
      "hotDesks": Object {
        "availableFrom": 2020-01-01T00:00:00.000Z,
        "currencyCode": "NOK",
        "desks": 0,
        "frequency": "Monthly",
        "minLease": 0,
        "price": 0,
      },
      "id": "0",
      "images": Array [],
      "months": undefined,
      "servicedOffices": Array [],
      "spaceDescription": "",
      "spaceFloor": undefined,
      "spaceFloored": "",
      "spaceHighlights": "",
      "spaceMatterPort": "",
      "spaceName": "",
      "spaceRent": undefined,
      "spaceServices": undefined,
      "spaceServicesNotNegotiable": false,
      "spaceSize": undefined,
      "spaceSizeCommon": undefined,
      "status": 0,
      "type": "",
      "use": "OFFICE",
    },
  },
  "subdistrict": Object {
    "id": 0,
    "name": "",
  },
}
`;

exports[`Onboarding reducer should return the initial state 2`] = `
Object {
  "address": "",
  "amenities": Object {},
  "buildingId": undefined,
  "city": "",
  "country": "",
  "description": "desc1",
  "district": "",
  "docs": Array [],
  "images": Array [],
  "isNew": undefined,
  "isPublishing": false,
  "latLng": undefined,
  "location": "",
  "name": "name1",
  "postCode": "",
  "spaceCount": 1,
  "spaces": Object {
    "0": Object {
      "availabilityFixed": 2020-01-01T00:00:00.000Z,
      "fitout": Array [],
      "fixedDesks": Object {
        "availableFrom": 2020-01-01T00:00:00.000Z,
        "currencyCode": "NOK",
        "desks": 0,
        "frequency": "Monthly",
        "minLease": 0,
        "price": 0,
      },
      "floorPlan": Array [],
      "hotDesks": Object {
        "availableFrom": 2020-01-01T00:00:00.000Z,
        "currencyCode": "NOK",
        "desks": 0,
        "frequency": "Monthly",
        "minLease": 0,
        "price": 0,
      },
      "id": "0",
      "images": Array [],
      "months": undefined,
      "servicedOffices": Array [],
      "spaceDescription": "",
      "spaceFloor": undefined,
      "spaceFloored": "",
      "spaceHighlights": "",
      "spaceMatterPort": "",
      "spaceName": "",
      "spaceRent": undefined,
      "spaceServices": undefined,
      "spaceServicesNotNegotiable": false,
      "spaceSize": undefined,
      "spaceSizeCommon": undefined,
      "status": 0,
      "type": "",
      "use": "OFFICE",
    },
  },
  "subdistrict": Object {
    "id": 0,
    "name": "",
  },
}
`;

exports[`Onboarding reducer should return the initial state 3`] = `
Object {
  "address": "addres1",
  "amenities": Object {
    "hasLifter": Object {
      "checked": true,
      "desc": "Lifert",
    },
    "hasSecurity": Object {
      "checked": false,
      "desc": "Security",
    },
    "none": Object {
      "checked": true,
      "desc": "none",
    },
  },
  "buildingId": "id1",
  "city": "Olso",
  "country": "NO",
  "defaultAmenities": Object {
    "hasGym": Object {
      "checked": true,
      "desc": "hasGym",
    },
  },
  "description": "descr1",
  "district": "",
  "docs": Array [],
  "error": "Error1",
  "images": Array [],
  "isNew": true,
  "isPublishing": true,
  "latLng": Object {
    "lat": 0,
    "lng": 0,
  },
  "location": "Olso",
  "name": "name1",
  "postCode": "Olso",
  "spaceCount": 2,
  "spaces": Object {
    "0": Object {
      "availabilityFixed": 2020-01-01T00:00:00.000Z,
      "fitout": Array [],
      "fixedDesks": Object {
        "availableFrom": 2020-01-01T00:00:00.000Z,
        "desks": 20,
        "minLease": 2,
        "price": 2000,
      },
      "floorPlan": Array [],
      "hotDesks": Object {
        "availableFrom": 2020-01-01T00:00:00.000Z,
        "desks": 20,
        "minLease": 2,
        "price": 2000,
      },
      "id": "0",
      "images": Array [],
      "months": 1,
      "servicedOffices": Object {
        "availableFrom": 2020-01-01T00:00:00.000Z,
        "desks": 20,
        "minLease": 2,
        "price": 2000,
      },
      "spaceDescription": "space1",
      "spaceFloor": 1,
      "spaceFloored": "data",
      "spaceHighlights": "data",
      "spaceMatterPort": "data",
      "spaceName": "space1",
      "spaceRent": 5000,
      "spaceServices": 500,
      "spaceServicesNotNegotiable": true,
      "spaceSize": 500,
      "spaceSizeCommon": 550,
      "status": 0,
      "type": "FLEX",
      "uploadedFloorPlans": Array [],
      "uploadedImages": Array [],
      "use": "OFFICE",
    },
    "1": Object {
      "availabilityFixed": 2020-01-01T00:00:00.000Z,
      "fitout": Array [],
      "fixedDesks": Object {
        "availableFrom": 2020-01-01T00:00:00.000Z,
        "currencyCode": "NOK",
        "desks": 0,
        "frequency": "Monthly",
        "minLease": 0,
        "price": 0,
      },
      "floorPlan": Array [],
      "hotDesks": Object {
        "availableFrom": 2020-01-01T00:00:00.000Z,
        "currencyCode": "NOK",
        "desks": 0,
        "frequency": "Monthly",
        "minLease": 0,
        "price": 0,
      },
      "id": "1",
      "images": Array [],
      "months": 1,
      "servicedOffices": Array [],
      "spaceDescription": "space1",
      "spaceFloor": 1,
      "spaceFloored": "data",
      "spaceHighlights": "data",
      "spaceMatterPort": "data",
      "spaceName": "space1",
      "spaceRent": 5000,
      "spaceServices": 500,
      "spaceServicesNotNegotiable": false,
      "spaceSize": 500,
      "spaceSizeCommon": 550,
      "status": 0,
      "type": "FLEX",
      "use": "OFFICE",
    },
  },
  "subdistrict": Object {
    "id": 0,
    "name": "",
  },
  "uploadedDocs": Array [],
  "uploadedImages": Array [],
}
`;

exports[`Onboarding reducer should return the initial state 4`] = `
Object {
  "address": "addres1",
  "amenities": Object {
    "hasLifter": Object {
      "checked": true,
      "desc": "Lifert",
    },
    "hasSecurity": Object {
      "checked": false,
      "desc": "Security",
    },
    "none": Object {
      "checked": true,
      "desc": "none",
    },
  },
  "buildingId": "id1",
  "city": "Olso",
  "country": "NO",
  "defaultAmenities": Object {
    "hasGym": Object {
      "checked": true,
      "desc": "hasGym",
    },
  },
  "description": "descr1",
  "district": "",
  "docs": Array [],
  "error": "Error1",
  "images": Array [],
  "isNew": true,
  "isPublishing": true,
  "latLng": Object {
    "lat": 0,
    "lng": 0,
  },
  "location": "Olso",
  "name": "name1",
  "postCode": "Olso",
  "spaceCount": 3,
  "spaces": Object {
    "0": Object {
      "availabilityFixed": 2020-01-01T00:00:00.000Z,
      "fitout": Array [],
      "fixedDesks": Object {
        "availableFrom": 2020-01-01T00:00:00.000Z,
        "currencyCode": "NOK",
        "desks": 0,
        "frequency": "Monthly",
        "minLease": 0,
        "price": 0,
      },
      "floorPlan": Array [],
      "hotDesks": Object {
        "availableFrom": 2020-01-01T00:00:00.000Z,
        "currencyCode": "NOK",
        "desks": 0,
        "frequency": "Monthly",
        "minLease": 0,
        "price": 0,
      },
      "id": "0",
      "images": Array [],
      "months": undefined,
      "servicedOffices": Array [],
      "spaceDescription": "",
      "spaceFloor": undefined,
      "spaceFloored": "",
      "spaceHighlights": "",
      "spaceMatterPort": "",
      "spaceName": "",
      "spaceRent": undefined,
      "spaceServices": undefined,
      "spaceServicesNotNegotiable": false,
      "spaceSize": undefined,
      "spaceSizeCommon": undefined,
      "status": 0,
      "type": "",
      "use": "OFFICE",
    },
    "2": Object {
      "availabilityFixed": 2020-01-01T00:00:00.000Z,
      "fitout": Array [],
      "fixedDesks": Object {
        "availableFrom": 2020-01-01T00:00:00.000Z,
        "currencyCode": "NOK",
        "desks": 0,
        "frequency": "Monthly",
        "minLease": 0,
        "price": 0,
      },
      "floorPlan": Array [],
      "hotDesks": Object {
        "availableFrom": 2020-01-01T00:00:00.000Z,
        "currencyCode": "NOK",
        "desks": 0,
        "frequency": "Monthly",
        "minLease": 0,
        "price": 0,
      },
      "id": "2",
      "images": Array [],
      "months": undefined,
      "servicedOffices": Array [],
      "spaceDescription": "",
      "spaceFloor": undefined,
      "spaceFloored": "",
      "spaceHighlights": "",
      "spaceMatterPort": "",
      "spaceName": "",
      "spaceRent": undefined,
      "spaceServices": undefined,
      "spaceServicesNotNegotiable": false,
      "spaceSize": undefined,
      "spaceSizeCommon": undefined,
      "status": 0,
      "type": "",
      "use": "OFFICE",
    },
  },
  "subdistrict": Object {
    "id": 0,
    "name": "",
  },
  "uploadedDocs": Array [],
  "uploadedImages": Array [],
}
`;

exports[`Onboarding reducer should return the initial state 5`] = `
Object {
  "address": "addres1",
  "amenities": Object {
    "hasLifter": Object {
      "checked": true,
      "desc": "Lifert",
    },
    "hasSecurity": Object {
      "checked": false,
      "desc": "Security",
    },
    "none": Object {
      "checked": true,
      "desc": "none",
    },
  },
  "buildingId": "id1",
  "city": "Olso",
  "country": "NO",
  "defaultAmenities": Object {
    "hasGym": Object {
      "checked": true,
      "desc": "hasGym",
    },
  },
  "description": "descr1",
  "district": "",
  "docs": Array [],
  "error": "Error1",
  "images": Array [],
  "isNew": true,
  "isPublishing": true,
  "latLng": Object {
    "lat": 0,
    "lng": 0,
  },
  "location": "Olso",
  "name": "name1",
  "postCode": "Olso",
  "spaceCount": 3,
  "spaces": Object {
    "2": Object {
      "availabilityFixed": 2020-01-01T00:00:00.000Z,
      "fitout": Array [],
      "fixedDesks": Object {
        "availableFrom": 2020-01-01T00:00:00.000Z,
        "currencyCode": "NOK",
        "desks": 0,
        "frequency": "Monthly",
        "minLease": 0,
        "price": 0,
      },
      "floorPlan": Array [],
      "hotDesks": Object {
        "availableFrom": 2020-01-01T00:00:00.000Z,
        "currencyCode": "NOK",
        "desks": 0,
        "frequency": "Monthly",
        "minLease": 0,
        "price": 0,
      },
      "id": "2",
      "images": Array [],
      "months": undefined,
      "servicedOffices": Array [],
      "spaceDescription": "",
      "spaceFloor": undefined,
      "spaceFloored": "",
      "spaceHighlights": "",
      "spaceMatterPort": "",
      "spaceName": "",
      "spaceRent": undefined,
      "spaceServices": undefined,
      "spaceServicesNotNegotiable": false,
      "spaceSize": undefined,
      "spaceSizeCommon": undefined,
      "status": 0,
      "type": "",
      "use": "OFFICE",
    },
  },
  "subdistrict": Object {
    "id": 0,
    "name": "",
  },
  "uploadedDocs": Array [],
  "uploadedImages": Array [],
}
`;
